frontend/simple/views/Invite.vue

```diff
@@ -86,105 +33,109 @@
+      'memberCount'
+    ])
+  },
+  methods: {
+    updateInvitees (payload) {
+      this.form.errorMsg = null
+      Object.assign(this.form, payload.data)
     },
     async submit () {
       try {
-        this.errorMsg = null
-        // TODO: as members are successfully invited display in a
-        // seperate invitees grid and add them to some validation for duplicate invites
-        for (let member of this.members) {
-          // We need to have the latest mailbox attribute for the user
-          const mailbox = await backend.latestHash(member.state.attributes.mailbox)
-          const sentDate = new Date().toString()
+        this.form.errorMsg = null
+        const groupId = this.currentGroupId
+        const groupName = this.currentGroupState.groupName
+
+        for (let member of this.form.invitees) {
+          let mailbox = member.state.attributes.mailbox
+          let memberName = member.state.attributes.name
 
-          // We need to post the invite to the users' mailbox contract
-          const invite = new Events.HashableMailboxPostMessage(
-            {
-              from: this.$store.getters.currentGroupState.groupName,
-              headers: [this.$store.state.currentGroupId],
-              messageType: Events.HashableMailboxPostMessage.TypeInvite,
-              sentDate
-            },
-            mailbox
-          )
-          await backend.publishLogEntry(member.state.attributes.mailbox, invite)
+          if (this.isAlreadyInvited(member)) {
+            this.form.errorMsg = template(L('Failed to invite users: member {name} is already invited'),
+              { name: memberName })
+            return
+          }
 
-          // We need to make a record of the invitation in the group's contract
-          const latest = await backend.latestHash(this.$store.state.currentGroupId)
-          const invited = new Events.HashableGroupRecordInvitation(
-            {
-              username: member.state.attributes.name,
-              inviteHash: invite.toHash(),
-              sentDate
-            },
-            latest
-          )
-          await backend.publishLogEntry(this.$store.state.currentGroupId, invited)
+          let inviteToMailbox = await sbp('proposals/invite/mailRecord/create', mailbox, groupName, groupId)
+          let inviteToGroup = await sbp('proposals/invite/groupRecord/create', inviteToMailbox.toHash(), memberName, groupId)
+
+          if (this.isProposal) {
+            let latest = await backend.latestHash(groupId)
+            let proposal = new Events.HashableGroupProposal({
+              // for proposal template selection in Vote.vue
+              type: 'invitationProposal', // TODO: constants for different proposal types
+              // calculate the voting threshold from the group data
+              // percentage: this.currentGroupState.memberApprovalPercentage * 0.01,
+              percentage: parseInt(this.currentGroupState.memberApprovalPercentage),
```

Comment:
I recommend renaming this to `votingThreshold`, make it a floating-point value so that you can multiply directly by it in the other places (instead of having to remember to `* 0.01`), and doing the pretty-ification (i.e. *user-display*) stuff in the user display area.

You can even use a [VueJS filter](https://vuejs.org/v2/guide/filters.html) to do the user-display stuff in the markup, e.g. `currentGroupState.memberApprovalThreshold | decimalToPercent` or something.

Reply:
yes, that totally makes sense! I spent way too much time trying to find the 3rd place where I needed the `* 0.01`-ing, so I feel this pain :)